http-clients/src/main/java/org/triplea/http/client/web/socket/WebSocketProtocolSwapper.java

```diff
@@ -0,0 +1,13 @@
+package org.triplea.http.client.web.socket;
+
+import java.net.URI;
+import java.util.function.Function;
+
+class WebSocketProtocolSwapper implements Function<URI, URI> {
```

Comment:
This class could even implement `UnaryOperator<URI>`
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/function/UnaryOperator.html